## Библиотека numpy

На массивах numpy строятся датафреймы - это тип данных (наши таблицы, но об этом позже)

Что интересного есть у numpy и что нам пригодится? 

Массив numpy напоминает собой список. И обычные списки мы можем переводить в массивы numpy и выполнять различные операции. 

Например, у нас есть список x=  [1,2,3,4], чтобы в списке умножить каждый элемент на 2, в питоне нам бы пришлось делать цикл for и тд. В numpy мы можем просто сделать x*2. Так работает с любой математической операцией. 

Это пригодится нам для решения некоторых задач уровня B. 

Чтобы создать массив numpy, пользуемся функцией array



In [2]:
x = [1,2,3,4]
x*2

[1, 2, 3, 4, 1, 2, 3, 4]

In [3]:
x = np.array([1,2,3,4])
x*2

array([2, 4, 6, 8])

In [1]:
import numpy as np
x = np.array([1,2,3,4])
y = np.array([3,4,5,6])
print((x-y)**2)

[4 4 4 4]


In [ ]:
+
-
* - умножение 
** - степень (4**2 = 16)
/ - деление 


## Библиотека pandas 

В библиотеке pandas есть два основных типа данных - pd.Series и pd.DataFrame. 

In [1]:
import pandas as pd
s = pd.Series(['a','b','c','d'])
s

0    a
1    b
2    c
3    d
dtype: object

In [12]:
b = {'столбец 1':['a', 'b', 'c'], 'столбец 2': ['d', 'e', 'f']}
b = pd.DataFrame(b)
b


столбец 1 столбец 2
0         a         d
1         b         e
2         c         f


Когда мы хотим анализировать данные, нам нужно взять таблицу (csv-файл), загрузить его в Юпитер, а затем попросить компьютер через pandas прочитать этот файл и превратить его в DataFrame.


In [1]:
import pandas as pd
# pd.read_excel('') - если открываем эксель
data = pd.read_csv('Maternal Health Risk Data Set.csv') 
data

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
0      25         130           80  15.0      98.0         86  high risk
1      35         140           90  13.0      98.0         70  high risk
2      29          90           70   8.0     100.0         80  high risk
3      30         140           85   7.0      98.0         70  high risk
4      35         120           60   6.1      98.0         76   low risk
...   ...         ...          ...   ...       ...        ...        ...
1009   22         120           60  15.0      98.0         80  high risk
1010   55         120           90  18.0      98.0         60  high risk
1011   35          85           60  19.0      98.0         86  high risk
1012   43         120           90  18.0      98.0         70  high risk
1013   32         120           65   6.0     101.0         76   mid risk

[1014 rows x 7 columns]

Age, SystolicBP, DiastolicBP, BS и т.д. - это столбцы (названия столбцов - так называемые индексы). По-другому столбцы называют переменными!

Строк в нашем датасете - 1014. Каждая строка имеет свой индекс от 0 до 1013. 

Чтобы анализировать данные, мы должна работать напрямую со столбцами или строками. Для этого нужно их индексировать. 

### Индексация 

In [5]:
# по столбцу
data['SystolicBP']

0       130
1       140
2        90
3       140
4       120
       ... 
1009    120
1010    120
1011     85
1012    120
1013    120
Name: SystolicBP, Length: 1014, dtype: int64

In [5]:
# несколько столбцов, 2 и больше
data[['Age', 'BS']]

Age    BS
0      25  15.0
1      35  13.0
2      29   8.0
3      30   7.0
4      35   6.1
...   ...   ...
1009   22  15.0
1010   55  18.0
1011   35  19.0
1012   43  18.0
1013   32   6.0

[1014 rows x 2 columns]

In [9]:
# определенные строки 
data[0:2]

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
0   25         130           80  15.0      98.0         86  high risk
1   35         140           90  13.0      98.0         70  high risk

### Фильтрация 

Когда мы хотим отсмотреть определенные данные в таблице, например, сколько людей старше 60 лет и т.д.

In [ ]:
Для фильтрации нужно поставить условие. Условия как и в питоне.
1. >
2. <
3. >=
4. <=
5. ==
6. !=

Можно ставить не только простые условия, но и сложные. Например, старше 30 лет, но младше 50. Для этого нужны операнды.

| - операнд "или" (если хотя бы выполняется одно из условий)

& - операнд "и" (должны выполняться оба условия)

In [95]:
# первый способ
data['Age'] > 40

0       False
1       False
2       False
3       False
4       False
        ...  
1009    False
1010     True
1011    False
1012     True
1013    False
Name: Age, Length: 1014, dtype: bool

In [16]:
(data['Age'] > 40).sum()

215

In [2]:
# второй способ
data[data['Age'] > 40]

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
9      42         130           80  18.0      98.0         70  high risk
14     48         120           80  11.0      98.0         88   mid risk
16     50         140           90  15.0      98.0         90  high risk
21     50         140           80   6.7      98.0         70   mid risk
28     49         120           90   7.2      98.0         77   low risk
...   ...         ...          ...   ...       ...        ...        ...
1002   55         140           95  19.0      98.0         77  high risk
1003   50         130          100  16.0      98.0         76  high risk
1008   48         120           80  11.0      98.0         88  high risk
1010   55         120           90  18.0      98.0         60  high risk
1012   43         120           90  18.0      98.0         70  high risk

[215 rows x 7 columns]

In [6]:
data[data['Age'] > 40]['SystolicBP']

9       130
14      120
16      140
21      140
28      120
       ... 
1002    140
1003    130
1008    120
1010    120
1012    120
Name: SystolicBP, Length: 215, dtype: int64

In [3]:
# отфильтрованную таблицу можно присвоить другой переменной 
df = data[data['Age'] > 40]

In [4]:
df['SystolicBP'].max()

140

In [97]:
(data['Age'] > 30)&(data['Age'] < 50)

0       False
1        True
2       False
3       False
4        True
        ...  
1009    False
1010    False
1011     True
1012     True
1013     True
Name: Age, Length: 1014, dtype: bool

In [41]:
# со сложным условием
data[(data['Age'] > 30) & (data['Age'] < 50)]

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
1      35         140           90  13.0      98.0         70  high risk
4      35         120           60   6.1      98.0         76   low risk
7      35          85           60  11.0     102.0         86  high risk
8      32         120           90   6.9      98.0         70   mid risk
9      42         130           80  18.0      98.0         70  high risk
...   ...         ...          ...   ...       ...        ...        ...
1007   40         120           75   7.7      98.0         70  high risk
1008   48         120           80  11.0      98.0         88  high risk
1011   35          85           60  19.0      98.0         86  high risk
1012   43         120           90  18.0      98.0         70  high risk
1013   32         120           65   6.0     101.0         76   mid risk

[260 rows x 7 columns]

In [18]:
data[(data['RiskLevel'] == 'high risk') | (data['RiskLevel'] == 'mid risk')]

Age  SystolicBP  DiastolicBP     BS  BodyTemp  HeartRate  RiskLevel  \
0      25         130           80  15.00      98.0         86  high risk   
1      35         140           90  13.00      98.0         70  high risk   
2      29          90           70   8.00     100.0         80  high risk   
3      30         140           85   7.00      98.0         70  high risk   
5      23         140           80   7.01      98.0         70  high risk   
...   ...         ...          ...    ...       ...        ...        ...   
1009   22         120           60  15.00      98.0         80  high risk   
1010   55         120           90  18.00      98.0         60  high risk   
1011   35          85           60  19.00      98.0         86  high risk   
1012   43         120           90  18.00      98.0         70  high risk   
1013   32         120           65   6.00     101.0         76   mid risk   

        sum     risk  
0     184.0     Risk  
1     168.0     Risk  
2     180.0     Risk  
3     168.0  No risk  
5     168.0     Risk  
...     ...      ...  
1009  178.0     Risk  
1010  158.0     Risk  
1011  184.0     Risk  
1012  168.0     Risk  
1013  177.0  No risk  

[608 rows x 9 columns]

Интересный лайфхак, который пригодится на экзамене. Допустим, нам нужно найти максимальный возраст пациентов, у которого уровень сахара равен 6. 

In [6]:
# Двойная индексация
data[data['BS'] == 6]['Age'].max()

32

### Сортировка 

In [5]:
data

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
0      25         130           80  15.0      98.0         86  high risk
1      35         140           90  13.0      98.0         70  high risk
2      29          90           70   8.0     100.0         80  high risk
3      30         140           85   7.0      98.0         70  high risk
4      35         120           60   6.1      98.0         76   low risk
...   ...         ...          ...   ...       ...        ...        ...
1009   22         120           60  15.0      98.0         80  high risk
1010   55         120           90  18.0      98.0         60  high risk
1011   35          85           60  19.0      98.0         86  high risk
1012   43         120           90  18.0      98.0         70  high risk
1013   32         120           65   6.0     101.0         76   mid risk

[1014 rows x 7 columns]

In [21]:
data.sort_values('Age')

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
849   10         100           50   6.0      99.0         70   mid risk
19    10          70           50   6.9      98.0         70   low risk
670   10         100           50   6.0      99.0         70   mid risk
250   10          85           65   6.9      98.0         70   low risk
588   12          90           60   7.5     102.0         66   mid risk
..   ...         ...          ...   ...       ...        ...        ...
435   65         130           80  15.0      98.0         86  high risk
339   65         120           90   6.9     103.0         76   low risk
322   65          90           60   6.9      98.0         70   low risk
326   66          85           60   6.9      98.0         86   low risk
338   70          85           60   6.9     102.0         70   low risk

[1014 rows x 7 columns]

In [8]:
data.sort_values(['Age', 'BS'], ascending=[False, True])

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
338    70          85           60   6.9     102.0         70   low risk
326    66          85           60   6.9      98.0         86   low risk
322    65          90           60   6.9      98.0         70   low risk
339    65         120           90   6.9     103.0         76   low risk
435    65         130           80  15.0      98.0         86  high risk
...   ...         ...          ...   ...       ...        ...        ...
1000   12          90           60  11.0     102.0         60  high risk
670    10         100           50   6.0      99.0         70   mid risk
849    10         100           50   6.0      99.0         70   mid risk
19     10          70           50   6.9      98.0         70   low risk
250    10          85           65   6.9      98.0         70   low risk

[1014 rows x 7 columns]

In [31]:
# по убыванию 
data.sort_values('Age', ascending = False)

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
338   70          85           60   6.9     102.0         70   low risk
326   66          85           60   6.9      98.0         86   low risk
322   65          90           60   6.9      98.0         70   low risk
435   65         130           80  15.0      98.0         86  high risk
339   65         120           90   6.9     103.0         76   low risk
..   ...         ...          ...   ...       ...        ...        ...
940   12          90           60   7.5     102.0         60   low risk
19    10          70           50   6.9      98.0         70   low risk
670   10         100           50   6.0      99.0         70   mid risk
849   10         100           50   6.0      99.0         70   mid risk
250   10          85           65   6.9      98.0         70   low risk

[1014 rows x 7 columns]

In [15]:
data['Age'].sort_values()

849    10
19     10
670    10
250    10
588    12
       ..
435    65
339    65
322    65
326    66
338    70
Name: Age, Length: 1014, dtype: int64

In [105]:
data['Age'].sort_values(ascending=False)

338    70
326    66
322    65
435    65
339    65
       ..
940    12
19     10
670    10
849    10
250    10
Name: Age, Length: 1014, dtype: int64

### Методы и другие полезные штуки

Как добавить новый столбец:

+

-

/

*

In [13]:
data

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel  \
0      25         130           80  15.0      98.0         86  high risk   
1      35         140           90  13.0      98.0         70  high risk   
2      29          90           70   8.0     100.0         80  high risk   
3      30         140           85   7.0      98.0         70  high risk   
4      35         120           60   6.1      98.0         76   low risk   
...   ...         ...          ...   ...       ...        ...        ...   
1009   22         120           60  15.0      98.0         80  high risk   
1010   55         120           90  18.0      98.0         60  high risk   
1011   35          85           60  19.0      98.0         86  high risk   
1012   43         120           90  18.0      98.0         70  high risk   
1013   32         120           65   6.0     101.0         76   mid risk   

        sum    new  
0     184.0  False  
1     168.0   True  
2     180.0  False  
3     168.0   True  
4     174.0  False  
...     ...    ...  
1009  178.0  False  
1010  158.0   True  
1011  184.0  False  
1012  168.0   True  
1013  177.0  False  

[1014 rows x 9 columns]

In [11]:
data['DiastolicBP'] > data['HeartRate']

0       False
1        True
2       False
3        True
4       False
        ...  
1009    False
1010     True
1011    False
1012     True
1013    False
Length: 1014, dtype: bool

In [9]:
# векторизация
data['sum'] = data['BodyTemp'] + data['HeartRate']

In [12]:
data['new'] = data['DiastolicBP'] > data['HeartRate']

In [14]:
data['new'].sum()

553

In [15]:
data['sum'].sum()

252

In [17]:
data['sum'].sum()

252

In [ ]:
+ - 
* - умножение
/ - деление 
** - возведение в степень

In [21]:
# кодирование
# lambda -  одноразовая функция 
data['risk'] = data['BS'].apply(lambda x: 'Risk' if x>7 else 'No risk')

In [16]:
def risk(BS):
    if BS>=7:
        return 'Risk'
    elif BS == 6:
        return 'скоро будет риск'
    else:
        return 'No risk'

In [16]:
# кодирование
data['risk'] = data['BS'].apply(risk)

In [17]:
data

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel  \
0      25         130           80  15.0      98.0         86  high risk   
1      35         140           90  13.0      98.0         70  high risk   
2      29          90           70   8.0     100.0         80  high risk   
3      30         140           85   7.0      98.0         70  high risk   
4      35         120           60   6.1      98.0         76   low risk   
...   ...         ...          ...   ...       ...        ...        ...   
1009   22         120           60  15.0      98.0         80  high risk   
1010   55         120           90  18.0      98.0         60  high risk   
1011   35          85           60  19.0      98.0         86  high risk   
1012   43         120           90  18.0      98.0         70  high risk   
1013   32         120           65   6.0     101.0         76   mid risk   

        new              risk  
0     False              Risk  
1      True              Risk  
2     False              Risk  
3      True              Risk  
4     False           No risk  
...     ...               ...  
1009  False              Risk  
1010   True              Risk  
1011  False              Risk  
1012   True              Risk  
1013  False  скоро будет риск  

[1014 rows x 9 columns]

In [71]:
data[data['risk'] == 'No risk'].shape

(21, 9)

In [ ]:
data['Platform_coded'] = data['Platform'].apply(lambda x: 1 if x=='PS3' else 0)

In [ ]:
ps 3 - 1
ps 4 - 2
все остальное - 0

In [ ]:
def coded(x):
    if x=='PS3':
        return 1
    elif x=='PS4':
        return 2
    else:
        return 0
data['Platfrom_coded'] = data['Platform'].apply(coded)

In [6]:
def coded(x):
    if x>=15:
        return ' big risk'
    elif x>6:
        return 'risk'
    else:
        return 'no risk'
data['new_risk'] = data['BS'].apply(coded)

In [7]:
df = data[data['Platfrom_coded']==1]

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel  \
0      25         130           80  15.0      98.0         86  high risk   
1      35         140           90  13.0      98.0         70  high risk   
2      29          90           70   8.0     100.0         80  high risk   
3      30         140           85   7.0      98.0         70  high risk   
4      35         120           60   6.1      98.0         76   low risk   
...   ...         ...          ...   ...       ...        ...        ...   
1009   22         120           60  15.0      98.0         80  high risk   
1010   55         120           90  18.0      98.0         60  high risk   
1011   35          85           60  19.0      98.0         86  high risk   
1012   43         120           90  18.0      98.0         70  high risk   
1013   32         120           65   6.0     101.0         76   mid risk   

         risk   new_risk  
0        Risk   big risk  
1        Risk       risk  
2        Risk       risk  
3        Risk       risk  
4        Risk       risk  
...       ...        ...  
1009     Risk   big risk  
1010     Risk   big risk  
1011     Risk   big risk  
1012     Risk   big risk  
1013  No risk    no risk  

[1014 rows x 9 columns]

In [12]:
# удаление столбца
del data['sum']

In [21]:
# метод показывающий размерность: строки и столбцы 
data[data['Age'] > 30].shape

(398, 7)

In [116]:
# начало
data.head()

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel  risk
0   25         130           80  15.0      98.0         86  high risk  Risk
1   35         140           90  13.0      98.0         70  high risk  Risk
2   29          90           70   8.0     100.0         80  high risk  Risk
3   30         140           85   7.0      98.0         70  high risk  Risk
4   35         120           60   6.1      98.0         76   low risk  Risk

In [63]:
data.head(10)

Age  SystolicBP  DiastolicBP     BS  BodyTemp  HeartRate  RiskLevel  risk
0   25         130           80  15.00      98.0         86  high risk  Risk
1   35         140           90  13.00      98.0         70  high risk  Risk
2   29          90           70   8.00     100.0         80  high risk  Risk
3   30         140           85   7.00      98.0         70  high risk  Risk
4   35         120           60   6.10      98.0         76   low risk  Risk
5   23         140           80   7.01      98.0         70  high risk  Risk
6   23         130           70   7.01      98.0         78   mid risk  Risk
7   35          85           60  11.00     102.0         86  high risk  Risk
8   32         120           90   6.90      98.0         70   mid risk  Risk
9   42         130           80  18.00      98.0         70  high risk  Risk

In [117]:
data.tail()

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel  risk
1009   22         120           60  15.0      98.0         80  high risk  Risk
1010   55         120           90  18.0      98.0         60  high risk  Risk
1011   35          85           60  19.0      98.0         86  high risk  Risk
1012   43         120           90  18.0      98.0         70  high risk  Risk
1013   32         120           65   6.0     101.0         76   mid risk  Risk

In [65]:
# конец 
data.tail(7)

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel  \
1007   40         120           75   7.7      98.0         70  high risk   
1008   48         120           80  11.0      98.0         88  high risk   
1009   22         120           60  15.0      98.0         80  high risk   
1010   55         120           90  18.0      98.0         60  high risk   
1011   35          85           60  19.0      98.0         86  high risk   
1012   43         120           90  18.0      98.0         70  high risk   
1013   32         120           65   6.0     101.0         76   mid risk   

         risk  
1007     Risk  
1008     Risk  
1009     Risk  
1010     Risk  
1011     Risk  
1012     Risk  
1013  No risk

In [119]:
# общая информация 
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1014 entries, 0 to 1013
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Age          1014 non-null   int64  
 1   SystolicBP   1014 non-null   int64  
 2   DiastolicBP  1014 non-null   int64  
 3   BS           1014 non-null   float64
 4   BodyTemp     1014 non-null   float64
 5   HeartRate    1014 non-null   int64  
 6   RiskLevel    1014 non-null   object 
 7   risk         1014 non-null   object 
dtypes: float64(2), int64(4), object(2)
memory usage: 63.5+ KB


In [3]:
data['Age'].isna().sum()

0

In [ ]:
NaN

In [16]:
data['RiskLevel'].value_counts()

low risk     406
mid risk     336
high risk    272
Name: RiskLevel, dtype: int64

In [17]:
data['Age'].value_counts()

23    71
19    67
17    63
15    60
35    50
25    48
32    48
22    45
50    43
29    40
21    38
40    37
55    37
12    35
60    34
30    30
28    28
48    26
42    23
20    22
31    22
18    19
16    16
54    12
13    12
27     9
49     9
45     9
37     6
33     5
34     5
43     5
10     4
36     4
39     4
24     3
65     3
44     3
14     3
26     3
63     2
38     2
59     2
62     1
51     1
66     1
56     1
70     1
41     1
46     1
Name: Age, dtype: int64

In [19]:
data['Age'].max()

70

In [30]:
data['Age'].min()

10

In [31]:
data['Age'].sum()

30290

In [26]:
sum(data['Age'])

30290

In [17]:
data = pd.read_csv('mcdonalds.csv')
data

category                                               item  \
0             Breakfast                                       Egg McMuffin   
1             Breakfast                                  Egg White Delight   
2             Breakfast                                   Sausage McMuffin   
3             Breakfast                          Sausage McMuffin with Egg   
4             Breakfast                   Sausage McMuffin with Egg Whites   
..                  ...                                                ...   
255  Smoothies & Shakes                 McFlurry with Oreo Cookies (Small)   
256  Smoothies & Shakes                McFlurry with Oreo Cookies (Medium)   
257  Smoothies & Shakes                 McFlurry with Oreo Cookies (Snack)   
258  Smoothies & Shakes  McFlurry with Reese's Peanut Butter Cups (Medium)   
259  Smoothies & Shakes   McFlurry with Reese's Peanut Butter Cups (Snack)   

     calories  total_fat  calcium  sugars  protein  
0         300         20       25       3       17  
1         250         12       25       3       18  
2         370         35       25       2       14  
3         450         43       30       2       21  
4         400         35       25       2       21  
..        ...        ...      ...     ...      ...  
255       510         26       40      64       12  
256       690         35       50      85       15  
257       340         17       25      43        8  
258       810         50       60     103       21  
259       410         25       30      51       10  

[260 rows x 7 columns]

In [18]:
# = in в питоне (что то есть в чем то )
data[data['item'].str.contains('Egg')].shape

(25, 7)

In [ ]:
.str.contains()

In [60]:
data['category'].drop_duplicates()

0               Breakfast
42            Beef & Pork
57         Chicken & Fish
84                 Salads
90         Snacks & Sides
103              Desserts
110             Beverages
137          Coffee & Tea
232    Smoothies & Shakes
Name: category, dtype: object

In [24]:
data['sugars'].drop_duplicates()

0        3
2        2
7        4
19      15
20      16
      ... 
251    115
252     89
253    128
256     85
258    103
Name: sugars, Length: 83, dtype: int64

In [23]:
# выводит количество уникальных элементов
data['Age'].nunique()

50

In [22]:
# выводит уникальные элементы
data['Age'].unique()

array([25, 35, 29, 30, 23, 32, 42, 19, 20, 48, 15, 50, 10, 40, 21, 18, 16,
       22, 49, 28, 12, 60, 55, 45, 31, 17, 26, 54, 44, 33, 13, 34, 38, 39,
       63, 14, 37, 51, 62, 43, 65, 66, 56, 70, 27, 36, 59, 24, 41, 46])

### Типы данных (переменных)

1. Количественные:
- дискретные (количество детей, год)
- непрерывные (возраст)
2. Качественные(категориальные):
- порядковые/ранговая(степени ожога)
- номинальные (цвет):
    - бинарные (пол, true/false)

1. Age: возраст, когда пациентка забеременела 
2. SystolicBP: верхнее значение артериального давления в мм рт. ст. 
3. DiastolicBP: нижнее значение артериального давления в мм рт. ст. 
4. BS: уровень глюкозы в крови в ммоль/л - 
5. HeartRate: нормальная частота сердечных сокращений в состоянии покоя в ударах в минуту - 
6. BodyTemp - температура тела по фаренгейту - 
7. Risk Level: прогнозируемый риск здоровью (1 - высокий, 2 - средний, 3 - низкий) 

##  Потренируемся

In [75]:
data

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel  \
0      25         130           80  15.0      98.0         86  high risk   
1      35         140           90  13.0      98.0         70  high risk   
2      29          90           70   8.0     100.0         80  high risk   
3      30         140           85   7.0      98.0         70  high risk   
4      35         120           60   6.1      98.0         76   low risk   
...   ...         ...          ...   ...       ...        ...        ...   
1009   22         120           60  15.0      98.0         80  high risk   
1010   55         120           90  18.0      98.0         60  high risk   
1011   35          85           60  19.0      98.0         86  high risk   
1012   43         120           90  18.0      98.0         70  high risk   
1013   32         120           65   6.0     101.0         76   mid risk   

         risk  
0        Risk  
1        Risk  
2        Risk  
3        Risk  
4        Risk  
...       ...  
1009     Risk  
1010     Risk  
1011     Risk  
1012     Risk  
1013  No risk  

[1014 rows x 8 columns]

1. Age: возраст, когда пациентка забеременела 
2. SystolicBP: верхнее значение артериального давления в мм рт. ст. 
3. DiastolicBP: нижнее значение артериального давления в мм рт. ст. 
4. BS: уровень глюкозы в крови в ммоль/л - 
5. HeartRate: нормальная частота сердечных сокращений в состоянии покоя в ударах в минуту - 
6. BodyTemp - температура тела по фаренгейту - 
7. Risk Level: прогнозируемый риск здоровью (1 - высокий, 2 - средний, 3 - низкий) 